benchmarking/src/lib.rs

```diff
@@ -1314,6 +1322,14 @@ macro_rules! add_benchmark {
 							.expect("benchmark name is always a valid string!")
 					);
 					None
+				},
+				Err($crate::BenchmarkError::Weightless) => {
+					$crate::log::error!(
+						"WARNING: benchmark weightless skipped - {}",
+						$crate::str::from_utf8(benchmark)
+							.expect("benchmark name is always a valid string!")
+					);
+					None
```

Comment:
```suggestion
					Some(vec![$crate::BenchmarkResult {
						components: selected_components.clone(),
						.. Default::default()
					}])
```

https://github.com/paritytech/substrate/pull/12829/files#diff-426f1e122298e2af0cd2d47a85bfacad3a9728d3bedfcee34545a8b062774c3fR1817-R1820